microsoft/terminal
# Provide the CloseWindow warning experience for the 'X' button

This is a follow up project of #1589 , which warns users of multiple tabs opened when press ALT+F4. 
Now we are going to connect this experience with the 'X' button of the window. 

For now, this PR only includes the implementation for IslandWindow (client window). Non-client window experience is still under implementation. 

 
## References


## PR Checklist


## Detailed Description of the Pull Request / Additional comments



To Test:

1. Start terminal. 
2. Open multiple tabs. 
3. Click 'X' on the top right corner of the window. 
4. You should be able  to see a warning. If you cancel it, then the Terminal keeps running. 
5. If you click 'Close', then the app will close. 
6. If you only have one tab opened, and click 'X', the app should close directly without any warnings.
